persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/dao/PostgresLockDAO.java

```diff
@@ -0,0 +1,130 @@
+package com.netflix.conductor.postgres.dao;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.conductor.core.sync.Lock;
+import com.netflix.conductor.model.TaskModel;
+import com.netflix.conductor.postgres.util.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.retry.support.RetryTemplate;
+
+import javax.sql.DataSource;
+import java.util.concurrent.*;
+
+public class PostgresLockDAO extends PostgresBaseDAO implements Lock {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresLockDAO.class);
+    private static final ConcurrentHashMap<String, ScheduledFuture<?>> SCHEDULEDFUTURES =
+            new ConcurrentHashMap<>();
+    private static final ThreadGroup THREAD_GROUP = new ThreadGroup("PostgresLock-scheduler");
+    private static final ThreadFactory THREAD_FACTORY =
+            runnable -> new Thread(THREAD_GROUP, runnable);
+    private static final ScheduledExecutorService SCHEDULER =
+            Executors.newScheduledThreadPool(1, THREAD_FACTORY);
+    protected PostgresLockDAO(RetryTemplate retryTemplate, ObjectMapper objectMapper, DataSource dataSource) {
+        super(retryTemplate, objectMapper, dataSource);
+    }
+
+    @Override
+    public void acquireLock(String lockId) {
+        LOGGER.trace("Acquiring lock {}", lockId);
+        // create code for acquiring advisory lock in postgres
+        final String ADVISORY_LOCK = "SELECT pg_advisory_lock (?)";
+
+        // execute the query
+        queryWithTransaction(ADVISORY_LOCK, (q) -> {
```

Comment:
we are using session level lock with `pg_advisory_lock` but using queryWithTransaction most likely uses random pooled sessions from some db connection pool in the background. Meaning that:

-  we would create the lock in random sessions
- also unlock in random / different sessions
- the pooled sessions can be closed at any time (releasing the locks)

we need to have a stable single session to execute all queries from this class (maybe have a dedicated connection pool with size 1 just for this) to make sure locking and unlocking works properly